emit `metadata` and `annotations`

jswrenn opened the pull request:
cc @bryangarza

jswrenn committed:
emit `metadata` and `annotations`

bryangarza commented on src/lib.rs:
```diff
@@ -11,29 +12,51 @@ use tracing_subscriber::registry::LookupSpan;
+/// X-Ray segment. Set the value of this field to either a
+/// [fresh trace id][trace_id::new], or one
+/// [parsed from HTTP headers][trace_id::from_headers].
+pub const TRACE_ID_FIELD: &'static str = "aws.xray.trace_id";
```
can you run clippy? i think it would warn about not needing static here because that is already implied by const

bryangarza replied:
can you run clippy and address the lints? for example i think it would warn about not needing static here because that is already implied by const

bryangarza commented on src/lib.rs:
```diff
@@ -109,16 +136,58 @@ pub mod trace_id {
+        SpanRef,
+    };
+
+    /// Extrace an AWS X-Ray `trace_id` from a tracing `Span`
```
typo?

bryangarza commented on src/lib.rs:
```diff
@@ -109,16 +136,58 @@ pub mod trace_id {
         })
     }
+    
+    use tracing_core::span::Attributes;
+    use tracing_core::subscriber::Subscriber;
+    use tracing_subscriber::registry::{
+        LookupSpan,
+        SpanRef,
+    };
```
can we move this to the top? or maybe this entire `trace_id` module to its own file and also move the imports up in that file?

bryangarza commented on src/lib.rs:
```diff
@@ -130,23 +199,44 @@ where
+        if let model::Kind::Segment = kind {
+            annotations
+                .fields
+                .add("tracing.name", attr.metadata().name());
```
can we make this a constant?

jswrenn replied:
I'm a little hesitant to make a change that increases indirection without decreasing repetition (since these are the only places these string literals occur).

bryangarza replied:
ok

bryangarza commented on src/lib.rs:
```diff
@@ -130,23 +199,44 @@ where
+        annotations
+            .fields
+            .add("tracing.target", attr.metadata().target());
+        metadata.fields.add("tracing.file", attr.metadata().file());
+        metadata.fields.add("tracing.line", attr.metadata().line());
```
let's make these constants too

bryangarza reviewed (changes requested):
Looks good, just left some smaller comments. Let's also do cargo fmt if we haven't already (I had done it for the daemon file but I didn't do it for the lib file)

bryangarza commented:
One other thing -- could you add a commit body? To explain in a sentence (or 2 or 3) what the change is. Makes it easier to go back through the git history later. I did something like that here https://github.com/jswrenn/tracing-xray/commit/e41af1f7708dae73148ba2b40e0f60ff6693d102.

jswrenn committed:
chore: clippy and refactoring

ran clippy
moved `trace_id` and `model` to distinct files
introduced `ANNOTATION_PREFIX` constant

jswrenn committed:
fix: missing `serialize_with` attribute on `end_time`
